java-squid/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -99,13 +113,21 @@ ProgramState clearStack() {
     for (int i = 0; i < nbElements; i++) {
       result.add(newStack.pop());
     }
-    return new Pair<>(new ProgramState(this, newStack), result);
+    return new Pop(new ProgramState(this, newStack), result);
   }
 
   public SymbolicValue peekValue() {
     return stack.peek();
   }
 
+  public List<SymbolicValue> peekValues(int n) {
+    if (n > stack.size()) {
+      throw new IllegalStateException("At least " + n + " values were expected on the stack!");
+    }
+    final ArrayList<SymbolicValue> result = new ArrayList<>(stack);
```

Comment:
No need to have a final local var, please use an Immutable List from guava and please build it looping through the stack as it will involves less iterations than constructing a new list and slicing it.